segmenters/nlp/NLTKSentencizer/__init__.py

```diff
@@ -0,0 +1,103 @@
+from typing import Dict, List
+
+from jina.executors.segmenters import BaseSegmenter
+
+
+class NLTKSentencizer(BaseSegmenter):
+    """
+    Segment text into sentences using :class:`nltk.PunktSentenceTokenizer`
+    for the specified language.
+
+    Example:
+
+    >>> sentencizer = NLTKSentencizer() # doctest:+ELLIPSIS
+    NLTKSentencizer@...[I]:post_init may take some time...
+    NLTKSentencizer@...[I]:NLTK english sentence tokenizer ready.
+    NLTKSentencizer@...[I]:post_init may take some time takes ... seconds (...s)
+    >>> text = "Today is a good day. Can't wait for tomorrow!"
+    >>> sentencizer.segment(text)
+    [{'text': 'Today is a good day.', 'offset': 0, 'location': [0, 20]}, {'text': "Can't wait for tomorrow!", 'offset': 1, 'location': [21, 45]}]
+
+
+    :param language: default='english'. Lowercased language name to initialize the sentence tokenizer, accepted languages are listed in :attr:`SUPPORTED_LANGUAGES`.
+    :type language: str
+    :param args: Additional positional arguments
+    :param kwargs: Additional keyword arguments
+    """
+
+    SUPPORTED_LANGUAGES = [
+        'czech',
+        'danish',
+        'dutch',
+        'english',
+        'estonian',
+        'finnish',
+        'french',
+        'german',
+        'greek',
+        'italian',
+        'norwegian',
+        'polish',
+        'portuguese',
+        'russian',
+        'slovene',
+        'spanish',
+        'swedish',
+        'turkish',
+    ]
+
+    def __init__(self, language: str = 'english', *args, **kwargs):
+        """Set constructor"""
+        super().__init__(*args, **kwargs)
+        self.language = language
+        self.sent_tokenizer = None
+
+    def post_init(self):
+        from nltk.data import load
+
+        try:
+            self.sent_tokenizer = load(f'tokenizers/punkt/{self.language}.pickle')
```

Comment:
why do not we pass a `tokenizer_path` as parameter to the object, from where to load the tokenizer object?

Reply:
I added tokenizer_path instead of sent_tokenizer to init, but I did not make it a parameter to the object. Because the nltk.data.load function only searches under nltk_data/tokenizers/punkt; otherwise throws an error. And path of nltk_data has to be one of the accepted paths, default being ~/nltk_data . So, it felt like it would not really be a parameter. Furthermore, the saved pickle file has to be of correct class etc. to load properly. 
Do you think it's ok to keep it like this?

Reply:
so u mean, the nltk will not accept an input path from another path? Why do they accept it as a parameter then?

Would it make sense to have it as Optional parameter  that defaults to the value you claim if not present?

Reply:
Yeah, you were right. I checked the load function, and default protocol searches that folder path but it's possible to specify with "file://..." . I am making the changes 👍